subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -644,21 +651,28 @@ private boolean allProjectsLocked() {
         return !projectLocks.isEmpty();
     }
 
-    private boolean canRunWithCurrentlyExecutedNodes(Node node, MutationInfo mutations) {
-        Set<String> candidateNodeDestroyables = mutations.destroyablePaths;
-
-        if (!runningNodes.isEmpty()) {
-            Set<String> candidateNodeOutputs = mutations.outputPaths;
-            Set<String> candidateMutations = !candidateNodeOutputs.isEmpty() ? candidateNodeOutputs : candidateNodeDestroyables;
-            if (hasNodeWithOverlappingMutations(candidateMutations)) {
+    private boolean canRunWithCurrentlyExecutedNodes(MutationInfo mutations) {
+        if (mutations.hasValidationProblem) {
+            if (!runningNodes.isEmpty()) {
+                // Invalid work is not allowed to run together with any other work
                 return false;
             }
+            invalidNodeRunning = true;
```

Comment:
The node won't necessarily start running here, since there is still a check afterwards: `doesDestroyNotYetConsumedOutputOfAnotherNode`

I think it would be better to flip this boolean when we are about to return the task we are going to run in `selectNextNode`. WDYT?

Reply:
Good point, I moved it a bit higher. I think it's in the right place now.